core/src/main/java/com/graphhopper/routing/util/DataFlagEncoder.java

```diff
@@ -575,9 +710,26 @@ public ConfigMap readStringMap(PMap weightingMap) {
 
         ConfigMap cMap = new ConfigMap();
         cMap.put("highways", map);
+
+        cloneDoubleAttribute(weightingMap, cMap, GenericWeighting.HEIGHT_LIMIT, 0d);
+        cloneDoubleAttribute(weightingMap, cMap, GenericWeighting.WEIGHT_LIMIT, 0d);
+        cloneDoubleAttribute(weightingMap, cMap, GenericWeighting.WIDTH_LIMIT, 0d);
+
         return cMap;
     }
 
+    private void cloneDoubleAttribute(PMap weightingMap, ConfigMap cMap, String key, double _default) {
+        if (weightingMap.has(key))
+            cMap.put(key, weightingMap.getDouble(key, _default));
+
+    }
+
+    protected void cloneStringAttribute(PMap weightingMap, ConfigMap cMap, String key, String _default) {
```

Comment:
I have a custom FlagEncoder in my project that subclasses DataFlagEncoder. I needed this method there, so make it 'protected'. The method is actually not used at all in DataFlagEncoder (didn't realize this). Will move the method to my subclass.